lifecycle-operator/controllers/lifecycle/keptnappversion/controller.go

```diff
@@ -220,3 +227,26 @@ func (r *KeptnAppVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *KeptnAppVersionReconciler) getTracer() telemetry.ITracer {
 	return r.TracerFactory.GetTracer(traceComponentName)
 }
+
+func (r *KeptnAppVersionReconciler) getLinkedSpans(appVersion *klcv1beta1.KeptnAppVersion) []trace.Link {
+	result := make([]trace.Link, len(appVersion.Spec.SpanLinks))
+
+	for i, linkedTrace := range appVersion.Spec.SpanLinks {
+		r.Log.Info("Adding Link to trace", "linkedTrace", linkedTrace)
+
+		traceContextCarrier := propagation.MapCarrier(map[string]string{
+			"traceparent": linkedTrace,
+		})
+
+		linkedCtx := context.Background()
+		linkedCtx = otel.GetTextMapPropagator().Extract(linkedCtx, traceContextCarrier)
+
+		link := trace.LinkFromContext(linkedCtx, attribute.KeyValue{
+			Key:   "ot-span-reference-type",
+			Value: attribute.StringValue("follows-from"),
+		})
+		result[i] = link
+
+	}
+	return result
```

Comment:
To stay consistent with the naming
```suggestion
	for i, linkedSpan := range appVersion.Spec.SpanLinks {
		r.Log.Info("Adding Link to span", "linkedSpan", linkedSpan)

		traceContextCarrier := propagation.MapCarrier(map[string]string{
			"traceparent": linkedSpan,
		})

		linkedCtx := context.Background()
		linkedCtx = otel.GetTextMapPropagator().Extract(linkedCtx, traceContextCarrier)

		link := trace.LinkFromContext(linkedCtx, attribute.KeyValue{
			Key:   "ot-span-reference-type",
			Value: attribute.StringValue("follows-from"),
		})
		result[i] = link

	}
	return result
```